Upgrade EUI to v70.2.4

cee-chen opened the pull request:
## Summary
`eui@67.1.8` ⏩ `eui@70.2.2`

⚠️ Note: This upgrade contains breaking changes to `EuiFlexGroup` and `EuiFlexGrid`, primarily around switching margins and negative margins to `gap`. Please do a quick QA pass of your app to scan for any issues. We're happy to help resolve minor fixes, or potentially follow up after PR merges. You can find us over in #eui!

## [`70.2.4`](https://github.com/elastic/eui/tree/v70.2.4)

**Bug fixes**

- Fixed visual bug in nested `EuiFlexGroup`s, where the parent `EuiFlexGroup` is responsive but a child `EuiFlexGroup` is not ([#6381](https://github.com/elastic/eui/pull/6381))

## [`70.2.3`](https://github.com/elastic/eui/tree/v70.2.3)

**Bug fixes**

- Fixed incorrect margins in `EuiSuperDatePicker` caused by `EuiFlex` CSS gap change ([#6380](https://github.com/elastic/eui/pull/6380))

## [`70.2.2`](https://github.com/elastic/eui/tree/v70.2.2)

- `EuiButton` now accepts `minWidth={false}` ([#6373](https://github.com/elastic/eui/pull/6373))

**Bug fixes**

- `EuiButton` no longer outputs unnecessary inline styles for `minWidth={0}` or `minWidth={false}` ([#6373](https://github.com/elastic/eui/pull/6373))
- `EuiFacetButton` no longer reports type issues when passing props accepted by `EuiButton` ([#6373](https://github.com/elastic/eui/pull/6373))
- Fixed the shadow sizes of `.eui-yScrollWithShadows` and `.eui-xScrollWithShadows` ([#6374](https://github.com/elastic/eui/pull/6374))


## [`70.2.1`](https://github.com/elastic/eui/tree/v70.2.1)

**Bug fixes**

- Re-fixed `EuiPageSection` not correctly merging `contentProps.css` ([#6365](https://github.com/elastic/eui/pull/6365))
- Fixed `EuiTab` not defaulting to size `m` ([#6366](https://github.com/elastic/eui/pull/6366))

## [`70.2.0`](https://github.com/elastic/eui/tree/v70.2.0)

- Added a keyboard shortcuts popover to `EuiDataGrid`'s toolbar. This can be visually hidden via `toolbarVisibility.showKeyboardShortcuts`, but will always remain accessible to keyboard and screen reader users. ([#6036](https://github.com/elastic/eui/pull/6036))
- `EuiScreenReaderOnly`'s `showOnFocus` prop now also shows on focus within its children ([#6036](https://github.com/elastic/eui/pull/6036))
- Added `onFocus` prop callback to `EuiSuperDatePicker` ([#6320](https://github.com/elastic/eui/pull/6320))

**Bug fixes**

- Fixed `EuiSelectable` to ensure the full options list is re-displayed when the search bar is controlled and cleared using `searchProps.value` ([#6317](https://github.com/elastic/eui/pull/6317))
- Fixed incorrect padding on `xl`-sized `EuiTabs` ([#6336](https://github.com/elastic/eui/pull/6336))
- Fixed `EuiCard` not correctly merging `css` on its child `icon`s ([#6341](https://github.com/elastic/eui/pull/6341))
- Fixed `EuiCheckableCard` not setting `css` on the correct DOM node ([#6341](https://github.com/elastic/eui/pull/6341))
- Fixed a webkit rendering issue with `EuiModal`s containing `EuiBasicTable`s tall enough to scroll ([#6343](https://github.com/elastic/eui/pull/6343))
- Fixed bug in `to_initials` that truncates custom initials ([#6346](https://github.com/elastic/eui/pull/6346))
- Fix bug in `EuiCard` where layout breaks when `horizontal` and `selectable` are both passed ([#6348](https://github.com/elastic/eui/pull/6348))

## [`70.1.0`](https://github.com/elastic/eui/tree/v70.1.0)

- Added the `hint` prop to the `<EuiSearchBar />`. This prop lets the consumer render a hint below the search bar that will be displayed on focus. ([#6319](https://github.com/elastic/eui/pull/6319))
- Added the `hasDragDrop` prop to `EuiPopover`. Use this prop if your popover contains `EuiDragDropContext`. ([#6329](https://github.com/elastic/eui/pull/6329))

**Bug fixes**

- Fixed `EuiButton`'s cursor style when the button is disabled ([#6323](https://github.com/elastic/eui/pull/6323))
- Fixed `EuiPageTemplate` not recognizing child `EuiPageSidebar`s/`EuiPageTemplate.Sidebar`s with `css` props ([#6324](https://github.com/elastic/eui/pull/6324))
- Fixed `EuiBetaBadge` to always respect its `anchorProps` values, including when there is no tooltip content ([#6326](https://github.com/elastic/eui/pull/6326))
- Temporarily patched `EuiModal` to not cause scroll-jumping issues on modal open ([#6327](https://github.com/elastic/eui/pull/6327))
- Fixed buggy drag & drop behavior within `EuiDataGrid`'s columns & sorting toolbar popovers ([#6329](https://github.com/elastic/eui/pull/6329))
- Fixed `EuiButton` not correctly passing `textProps` for children inside fragments or i18n components ([#6332](https://github.com/elastic/eui/pull/6332))
- Fixed `EuiButton` not correctly respecting `minWidth={0}` ([#6332](https://github.com/elastic/eui/pull/6332))

**CSS-in-JS conversions**

- Converted `EuiTabs` to Emotion ([#6311](https://github.com/elastic/eui/pull/6311))

## [`70.0.0`](https://github.com/elastic/eui/tree/v70.0.0)

- Added the `enabled` option to the `<EuiInMemoryTable />` `executeQueryOptions` prop. This option prevents the Query from being executed when controlled by the consumer. ([#6284](https://github.com/elastic/eui/pull/6284))

**Bug fixes**

- Fixed `EuiOverlayMask` to set a `[data-relative-to-header=above|below]` attribute to replace the `--aboveHeader` and `--belowHeader` classNames removed in its Emotion conversion ([#6289](https://github.com/elastic/eui/pull/6289))
- Fixed `EuiHeader` CSS using removed `EuiOverlayMask` class modifiers ([#6293](https://github.com/elastic/eui/pull/6293))
- Fixed `EuiToolTip` not respecting reduced motion preferences ([#6295](https://github.com/elastic/eui/pull/6295))
- Fixed a bug with `EuiTour` where passing any `panelProps` would cause the beacon to disappear ([#6298](https://github.com/elastic/eui/pull/6298))

**Breaking changes**

- `@emotion/css` is now a required peer dependency, alongside `@emotion/react` ([#6288](https://github.com/elastic/eui/pull/6288))
- `@emotion/cache` is no longer required peer dependency, although your project must still use it if setting custom cache/injection locations ([#6288](https://github.com/elastic/eui/pull/6288))

**CSS-in-JS conversions**

- Converted `EuiCode` and `EuiCodeBlock` to Emotion; Removed `euiCodeSyntaxTokens` Sass mixin and `$euiCodeBlockPaddingModifiers`; ([#6263](https://github.com/elastic/eui/pull/6263))
- Converted `EuiResizableContainer` and `EuiResizablePanel` to Emotion ([#6287](https://github.com/elastic/eui/pull/6287))

## [`69.0.0`](https://github.com/elastic/eui/tree/v69.0.0)

- Added support for `fullWidth` prop on EuiForm, which will be the default for all rows/controls within ([#6229](https://github.com/elastic/eui/pull/6229))
- Added support for `onResizeStart` and `onResizeEnd` callbacks to `EuiResizableContainer` ([#6236](https://github.com/elastic/eui/pull/6236))
- Added optional case sensitive option matching to `EuiComboBox` with the `isCaseSensitive` prop ([#6268](https://github.com/elastic/eui/pull/6268))
- `EuiFlexItem` now supports `grow={0}` ([#6270](https://github.com/elastic/eui/pull/6270))
- Added the `alignItems` prop to `EuiFlexGrid` ([#6281](https://github.com/elastic/eui/pull/6281))
- Added `filter`, `filterExclude`, `filterIgnore`, `filterInclude`, `indexTemporary`, `infinity`, `sortAscending`, and `sortDescending` glyphs to `EuiIcon` ([#6282](https://github.com/elastic/eui/pull/6282))

**Bug fixes**

- Fixed `EuiTextProps` to show the `color` type option `inherit` as default ([#6267](https://github.com/elastic/eui/pull/6267))
- `EuiFlexGroup` now correctly respects `gutterSize` when responsive ([#6270](https://github.com/elastic/eui/pull/6270))
- Fixed the last breadcrumb in `EuiBreadcrumbs`'s `breadcrumbs` array not respecting `truncate` overrides ([#6280](https://github.com/elastic/eui/pull/6280))

**Breaking changes**

- `EuiFlexGrid` no longer supports `columns={0}`. Use `EuiFlexGroup` instead for normal flex display ([#6270](https://github.com/elastic/eui/pull/6270))
- `EuiFlexGrid` now uses modern `display: grid` CSS ([#6270](https://github.com/elastic/eui/pull/6270))
- `EuiFlexGroup`, `EuiFlexGrid`, and `EuiFlexItem` now use modern `gap` CSS instead of margins and negative margins ([#6270](https://github.com/elastic/eui/pull/6270))
- `EuiFlexGroup` no longer applies responsive styles to `column` or `columnReverse` directions ([#6270](https://github.com/elastic/eui/pull/6270))

**CSS-in-JS conversions**

- Converted `EuiFlexGroup`, `EuiFlexGrid`, and `EuiFlexItem` to Emotion ([#6270](https://github.com/elastic/eui/pull/6270))

## [`68.0.0`](https://github.com/elastic/eui/tree/v68.0.0)

- Added `beta` glyph to `EuiIcon` ([#6250](https://github.com/elastic/eui/pull/6250))
- Added `launch` and `spaces` glyphs to `EuiIcon` ([#6260](https://github.com/elastic/eui/pull/6260))
- Added the `fallbackDestination` prop to `EuiSkipLink`, which accepts a string of query selectors to fall back to if the `destinationId` does not have a valid target. Defaults to `main` ([#6261](https://github.com/elastic/eui/pull/6261))
- `EuiSkipLink` is now always an `a` tag to ensure that it is always placed within screen reader link menus. ([#6261](https://github.com/elastic/eui/pull/6261))

**Bug fixes**

- Fixed `EuiSuperDatePicker` not correctly merging passed `className`s ([#6253](https://github.com/elastic/eui/pull/6253))
- Fixed `EuiColorStops` not correctly merging in passed `data-test-subj`s, `style`s, or `...rest` ([#6255](https://github.com/elastic/eui/pull/6255))
- Fixed `EuiResizablePanel` incorrectly passing `style` to the wrapper instead of the panel. Use `wrapperProps.style` to pass styles to the wrapper. ([#6255](https://github.com/elastic/eui/pull/6255))
- Fixed custom `onClick`s passed to `EuiSkipLink` overriding `overrideLinkBehavior` ([#6261](https://github.com/elastic/eui/pull/6261))

**Breaking changes**

- Removed `inherit` and `ghost` color from `EuiListGroupItem` ([#6207](https://github.com/elastic/eui/pull/6207))
- Changed default color to `text` instead of `inherit` ([#6207](https://github.com/elastic/eui/pull/6207))

**CSS-in-JS conversions**

- Converted `EuiListGroup` and `EuiListGroupItem` to Emotion; Removed `$euiListGroupGutterTypes`, `$euiListGroupItemColorTypes` and `$euiListGroupItemSizeTypes`; ([#6207](https://github.com/elastic/eui/pull/6207))
- Converted `EuiBadgeGroup` to Emotion ([#6258](https://github.com/elastic/eui/pull/6258))
- Converted `EuiBetaBadge` to Emotion ([#6258](https://github.com/elastic/eui/pull/6258))
- Converted `EuiNotificationBadge` to Emotion ([#6258](https://github.com/elastic/eui/pull/6258))

cee-chen committed:
Upgrade EUI to v70.2.0

cee-chen committed:
Address EUI tooltip TODO from previous EUI upgrade

cee-chen committed:
Address EUI TODO for selectable & horizontal EuiCards

- reverts temporary fixes; should now be addressed in EUI itself

cee-chen committed:
Update i18n tokens

cee-chen committed:
[EuiCode][EuiCodeBlock] Account for removed Sass variables

cee-chen committed:
[EuiFlex] Fix type issues

cee-chen committed:
[EuiListGroup][EuiCollapsibleNav] Replace removed `color="ghost"` with theme color mode

cee-chen committed:
[EuiSkipLink] Fix event type

cee-chen committed:
[EuiCodeBlock] Account for removed niche color var

- corresponds to visColors, so convert to that util instead

[refactor] while here, convert file to using native EUI `useEuiTheme` instead of referencing the Sass vars

cee-chen committed:
[EuiFlex] Fix removed Sass gutter variables

- replace with basic EUI size variables

cee-chen committed:
[EuiCodeBlock] Fix index management component using now-removed CSS classes

address TODO by using component directly - EuiCodeBlock now supports JSX/nodes, albeit without syntax highlighting (not an issue since that wasn't being used here)

cee-chen committed:
[EuiFlex] Snapshot updates

cee-chen committed:
[EuiFlex] Fix tests assertions

- Emotion adds another wrapper, so we now have to use `.last()` in Enzyme finding

- className modifiers have been removed, so we have to replace them with their Emotion equivalents - this isn't great but to be honest we don't recommend asserting on EUI classNames anyway

- certain selectors updated to specificy their raw/DOM output vs enzyme wrappers

cee-chen committed:
[EuiForm] Update snapshots caused by `fullWidth` support

cee-chen committed:
[EuiListGroup] Update snapshots

cee-chen committed:
[EuiCard] Update snapshots

cee-chen committed:
[EuiSwitch][EuiCheckbox] Update snapshots

cee-chen committed:
[EuiOverlayMask] Update snapshots

cee-chen committed:
[EuiTab] Fix selectors/test assertions

Emotion wrapper is now taking the `key` from EuiTab, so we can no longer use that selector. Swtiching to data-test-subj instead of `key` fixes the issue (although it creates type shenanigans)

cee-chen committed:
[EuiBadge] Update snapshots/tests

cee-chen committed:
[EuiDataGrid] Update fixed `data-test-subj` selector

cee-chen committed:
Fix multiple tests to use rendered vs mounted snapshots

- Emotion wrappers add a TON of cruft to mounted snapshots, so we're moving away from them

cee-chen committed:
Update visual diff tolerance

- NB: I can't figure out how to update baseline screenshots locally without making the diff even worse than just updating it. Guidance appreciated, or teams can go in later and update their baselines

cee-chen committed:
[FTR] Fix now-invalid selector caused by Emotion classes

`--selected` is no longer at the end of the className list, but exists in there, so switch to `*=`

cee-chen committed:
[EuiSkipLink] Fix custom usage for table

- prefer a similar-looking EuiButton instead of EuiSkipLink directly, since EuiSkipLink's scrolling and fallback behavior is not needed

copy EuiSkipLink's minor style overrides & positioning to SCSS instead

cee-chen committed:
[Security] Fix failing scroll test

- appears to be back to 0 this upgrade for whatever reason. Changing the test to < 5 should account for any variance in overflow/scrollbars

cee-chen committed:
[Security] Fix broken looking flyout tabs

- Use `append` instead of custom flex group within `name`

cee-chen committed:
[Security] Fix spacing above flyout tabs

- Appears to be present in previous iterations of the flyout, but was removed recently. Opinionated change but it does need some spacing

- NB: fragment is needed due to inherited `children` typing - it doesn't appear to want multiple children, despite it working fine in browser

cee-chen committed:
[EuiFlex] Fix CSS references to removed modifier classes

cee-chen committed:
[Discover] Fix broken tabs/badge alignment

cee-chen committed:
[Security] Fix hidden flyout/failing Cypress test caused by `styled-components` not overriding Emotion z-index CSS

- adding either !important or a single extra layer of specificity fixes the issue

cee-chen committed:
[EuiFlex] Replace removed fractions variables

not totally sure why these were even exported

cee-chen committed:
[osquery] attempt to fix failing tests

elasticmachine commented:
Pinging @elastic/eui-design (EUI)

YulNaumenko reviewed (approved):
Security Solution changes LGTM

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

MiriamAparicio commented on x-pack/plugins/apm/public/components/app/service_map/empty_banner.tsx:
```diff
@@ -14,11 +14,11 @@ import { useTheme } from '../../../hooks/use_theme';
 
 const EmptyBannerContainer = euiStyled.div`
-  margin: ${({ theme }) => theme.eui.gutterTypes.gutterSmall};
+  margin: ${({ theme }) => theme.eui.euiSizeXS};
```
Shouldn't be this one also `euiSizeS`?

constancecchen replied:
You're totally right, this one was a brain fart on my part!

MiriamAparicio commented on x-pack/plugins/apm/public/components/app/service_map/controls.tsx:
```diff
@@ -19,9 +19,9 @@ import { getAnimationOptions, getNodeHeight } from './cytoscape_options';
 
 const ControlsContainer = euiStyled('div')`
-  left: ${({ theme }) => theme.eui.gutterTypes.gutterMedium};
+  left: ${({ theme }) => theme.eui.euiSize};
```
Is euiSize medium? or there's a euiSizeM?

constancecchen replied:
`euiSize` is medium: https://github.com/elastic/eui/blob/ff73b1f80dcf7500d57862348c746161a6ec6b9b/src-docs/src/views/theme/_json/eui_theme_light.json#L333

There is no `euiSizeM` in our Sass variables / JS variables exported from our Sass via JSON, but there is in our new JS-first `euiTheme`, used in our CSS-in-JS. Our naming generally strives to be clearer in our new CSS-in-JS code than in our older Sass code!

cnasikas reviewed (approved):
ResponseOps changes LGTM

juliaElastic commented on x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/step_define_package_policy.tsx:
```diff
@@ -44,7 +44,7 @@ import { PackagePolicyInputVarField } from './components';
 const FormGroupResponsiveFields = styled(EuiDescribedFormGroup)`
   @media (max-width: 767px) {
-    .euiFlexGroup--responsive {
+    [class*='euiFlexGroup-responsive'] {
```
is this a change in styled components or only an alternative way of defining class?

constancecchen replied:
An alternate way of defining class. This selector change is due to EUI's output classes changing from our Emotion conversions. We're generally removing modifier classes (but keeping parent classNames like `.euiFlexGroup`), but technically those modifiers are still accessible via Emotion's generated class name concatenation:

<img width="816" alt="" src="https://user-images.githubusercontent.com/549407/201993916-38df8f2b-5b93-4ac5-a655-00f7eeff01b3.png">

This change is essentially a quick fix to maintain current working behavior - without it, your selector would no longer be working after this upgrade. In general, or for a more long-term fix, we recommend passing in your own custom classNames (conditionally if necessary), and recommend against targeting our state-based classNames (our classNames have never been a guaranteed API/hook).

pgayvallet commented on packages/core/chrome/core-chrome-browser-internal/src/ui/header/collapsible_nav.tsx:
```diff
@@ -162,35 +163,35 @@ export function CollapsibleNav({
-                size="s"
-                color="ghost"
-              />
+              <EuiThemeProvider colorMode="dark">
```
We're already providing the theme provider at the root level of the react tree, so why do you need to add another one here? Forcing colorMode to `dark` seems wrong to me?

constancecchen replied:
It's done because these types of sections in the `EuiCollapsibleNav` is **always** dark. You can see an example on our docs here: https://elastic.github.io/eui/#/navigation/collapsible-nav#nav-groups-with-lists-and-other-content

<img width="458" alt="" src="https://user-images.githubusercontent.com/549407/201991154-12b300a7-6a13-4026-8867-3a7ca9ec3dd8.png">

The reason why switching/forcing the colorMode is superior is twofold: it cascades to **all** child components - links, text color, buttons, icon fills, etc. - everything just switches without needing a bunch of custom cascading/nesting CSS.

`EuiBottomBar` also has the same logic / nested dark provider as this.

chandlerprall replied:
We also dropped support for `EuiListGroup[color=ghost]` as a design decision,

> Breaking changes
> * Removed ghost color from EuiListGroupItem. Consumers now must pass a EuiThemProvider color="dark" to use a ghost color.
> * Removed inherit color from EuiListGroupItem. The inherit in my opinion was not making sense anymore. The consumer should pass the color they think makes sense for their use case.
> * The list items now defaults to color text

and our upgrade strategy for Kibana historically defaults to _maintain the same look & feel_ unless we've pre-coordinated changes with affected teams, which is why we're adding the theme provider here instead of changing the UX away from the ghost color.

FYI this is prompting a wider discussion about these types of breaking changes in https://github.com/elastic/eui/discussions/6364

constancecchen replied:
@miukimiu mentioned this as well in our sync today, but I would be curious if people would feel less confused about this change if we preserved `color="ghost"` as a prop/API option and had the component automatically internally wrap `<EuiThemeProvider colorMode="dark">` instead of consumers explicitly having to.

pgayvallet replied:
Thanks for the explanations

thomheymann reviewed (approved):
security changes LGTM

jennypavlova reviewed (changes requested):
Good job! There is one small issue on the inventory page I spotted - the content padding is missing. 
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/14139027/201913078-78f82652-ac49-4e17-991f-b7e599558ae8.png">
I think it is related to the custom styles defined in `fullHeightContentStyles` which overrides the content styles. 
<img width="2482" alt="image" src="https://user-images.githubusercontent.com/14139027/201912760-e7743098-02cf-4658-93ca-6dd367b1edd6.png">

I checked in the browser and the content styles are missing (only the custom styles are visible). I remember we had a similar issue before and it was fixed in [64.0.4](https://github.com/elastic/kibana/pull/140323)

Can you please check that? Maybe we can also check if we should drop the custom styles and use them as optional styles in the `pageContent` itself if it's not possible to support them anymore 🤔

tomsonpl reviewed (approved):
Asset Management LGTM 👍

alisonelizabeth commented on x-pack/plugins/snapshot_restore/__jest__/client_integration/home.test.ts:
```diff
@@ -748,7 +748,7 @@ describe('<SnapshotRestoreHome />', () => {
 
               const tabs = find('snapshotDetail.tab');
-              const selectedTab = find('snapshotDetail').find('.euiTab-isSelected');
+              const selectedTab = find('snapshotDetail').find('.euiTab-isSelected').last();
```
Can you explain why the `.last()` is needed now?

chandlerprall replied:
Because there's an infinite ways to do anything in JS...

Enzyme's [`hostNodes`](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/hostNodes.html) can be a good alternative in these scenarios where you need to trim down to only "real" DOM elements. Discovered this method during the last upgrade, until now I've also been using `last()`.

alisonelizabeth reviewed (approved):
Deployment Management changes look good. Left one question around a test change.

xcrzx reviewed (commented):
I've tested this PR locally and found some layout issues in the detection rules area. @constancecchen Do you need help fixing them?

![Screenshot 2022-11-15 at 18 10 18](https://user-images.githubusercontent.com/1938181/201989018-2b40e151-dfd0-4eae-96e6-994245636d84.png)

![Screenshot 2022-11-15 at 17 23 34](https://user-images.githubusercontent.com/1938181/201989178-53e9b2ab-86bf-4ea6-995c-d98732bb0091.png)

![Screenshot 2022-11-15 at 17 20 21](https://user-images.githubusercontent.com/1938181/201989395-d6bf381a-eb31-4d7c-ae11-fae3a94b89a1.png)

Commit:
[APM][PR feedback] Fix size conversion var

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

cee-chen committed:
[Security] Fix incorrect use of EuiFlexGrid

- should be EuiFlexGroup instead if only one row is desired

cee-chen committed:
Upgrade EUI to 70.2.1 to get backport fixes

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

constancecchen commented:
@jennypavlova Thank you for catching this - this behavior was regressed in EUI 69.x and I've just merged a fix for it into the EUI repo. I've backported the fix to 70.2.x for this PR and it should be resolved now, if you would like to double check.

@xcrzx Thank you for the thorough QA! The tabs display was also a bug on our end from its Emotion conversion and a fix has been backported. The table header regression was an issue of using `EuiFlexGrid` instead of `EuiFlexGroup` and has been fixed in a commit in this PR (2dde3c19bafcb9b0fbc17c087a0da805dc8ef538).

The EuiDataGrid toolbar screenshot is slightly more tricky. In theory all you need to do is move your absolutely positioned toolbar more to the left to accommodate the new keyboard shortcuts button. However, I have to ask - why is your team absolutely positioning toolbar controls on top of EuiDataGrid as opposed to using the `additionalControls.right` API ([link](https://elastic.github.io/eui/#/tabular-content/data-grid-toolbar#additional-controls-in-the-toolbar))? I can take a shot at updating your data grid to use `additionalControls`, but since this isn't a quick fix / is a more nuanced issue, I would suggest opening up an issue and addressing in a follow-up PR instead of this one (this upgrade is not landing in 8.6 so you will have ample time before 8.7FF to solve it).

MiriamAparicio reviewed (approved):
APM UI changes LGTM

jennypavlova reviewed (approved):
@constancecchen Thank you for the fix. I checked and now it looks good! 🚀

xcrzx commented:
> The EuiDataGrid toolbar screenshot is slightly more tricky. In theory all you need to do is move your absolutely positioned toolbar more to the left to accommodate the new keyboard shortcuts button. However, I have to ask - why is your team absolutely positioning toolbar controls on top of EuiDataGrid as opposed to using the `additionalControls.right` API ([link](https://elastic.github.io/eui/#/tabular-content/data-grid-toolbar#additional-controls-in-the-toolbar))?

If I'm not mistaken, that component is owned by @elastic/security-threat-hunting-investigations. So it would be better to align with them on an appropriate fix.

As for the Rules Area-related issues, verified locally - everything looks good now. Thank you for fixing them 👍

mdefazio reviewed (commented):
I'm seeing the padding around the Cases stats is now gone.  Wondering if its bc of this change:
```EuiFlexGroup, EuiFlexGrid, and EuiFlexItem now use modern gap CSS instead of margins and negative margins (#6270)```

With the EUI PR:
<img width="1242" alt="image" src="https://user-images.githubusercontent.com/3756330/202174985-2beed2d9-82c9-4007-b640-f878086d0798.png">

Current:
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/3756330/202175354-dfd21215-ce62-4f26-8f29-74a87bf29450.png">

cc/ @cnasikas

stratoula commented:
On the unified search bar we used to have this icon but this PR slighly changes it. Is it on purpose?

Main
<img width="822" alt="image" src="https://user-images.githubusercontent.com/17003240/202194171-080ee505-c5ac-43ec-9dfd-c6a876aaa54e.png">

Here:

<img width="969" alt="image" src="https://user-images.githubusercontent.com/17003240/202194136-13caabd9-8769-4ad3-877f-29db212e8633.png">

stratoula commented:
Also in Lens the functions have been broken

main:
<img width="483" alt="image" src="https://user-images.githubusercontent.com/17003240/202195249-a888f5e2-8cd3-4125-99ce-2b463e1bfccf.png">


here: 

<img width="436" alt="image" src="https://user-images.githubusercontent.com/17003240/202195441-06aca0de-175e-4943-b398-e52747b4785f.png">

miukimiu commented:
> On the unified search bar we used to have this icon but this PR slightly changes it. Is it on purpose?

@stratoula yes, it is on purpose. The filter icon was updated by @MichaelMarcialis in https://github.com/elastic/eui/pull/6282 to support new Lens features.

peteharverson commented:
For ML, each item in the expanded rows (`EuiFlexGrid`) of the data visualizer is now on a separate row. Is there something that we need to change to revert to the previous behavior?

Before:
<img width="1484" alt="image" src="https://user-images.githubusercontent.com/7405507/202220781-0cfc33e2-f7ea-4479-a2d6-36956d391071.png">


After:
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/7405507/202220419-d2ec0807-ebd5-49bf-872d-b4ad228d9272.png">

ThomThomson reviewed (approved):
Snapshot and class name changes LGTM. Code only review this time.

cee-chen committed:
[Security][Cases] Fix padding previously reliant on EuiFlex margins

+ simplify CSS - these styles can be applied to the wrapper directly instead of nesting into the EuiFlexGroup

cee-chen committed:
[Lens] Restore EuiListGroup display back to previous custom columns

peteharverson commented:
Also found what looks like an `EuiFlexGroup` issue with the job groups control in ML:

Before:
<img width="526" alt="image" src="https://user-images.githubusercontent.com/7405507/202254494-3e1479dd-38cc-478c-b35c-475a01029bb7.png">

After:
<img width="344" alt="image" src="https://user-images.githubusercontent.com/7405507/202254592-461aa73a-6066-41ff-be0e-d52770a2e630.png">

cee-chen committed:
[ML][DataVisualizer] Fix incorrect usage of EuiFlexGrid

- should be EuiFlexGroup instead if 0 columns are being specified

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

constancecchen commented:
@peteharverson can you include information about the page/view you're on, ideally with a link to the URL/file(s) affected? I'm generally able fix most of the issues being posted if given a way to repro, but I'm not as familiar with y'all's individual products and apps as you are and could use help finding the UI in question!

I should have just fixed the expanded row table display in your earlier screenshot (ca332d2). Trying to figure out how to get to your jobs view now. There's two different Jobs links in the sidebar so it's not totally clear.

@stratoula Lens Functions columns should also be fixed now (9ca6ed5).

@mdefazio The cases flex group padding should also be fixed now (6dcdf94).

cee-chen committed:
[ML][Jobs] Fix missing margins in Jobs Groups list
- caused by switch to `gap`

opauloh reviewed (approved):
awp-viz changes lgtm

cee-chen committed:
Update snapshots from main

cee-chen committed:
more snapshots

nreese reviewed (commented):
Found 2 issues in maps.

In the screen shot below, notice how "All", "Reference", "Elasticsearch", and "Solutions" tags overflow to 2 lines instead of being on a single line? To view, open maps application, create new map, click "Add layer"
<img width="600" alt="Screen Shot 2022-11-16 at 1 32 07 PM" src="https://user-images.githubusercontent.com/373691/202288876-97776ed1-2dd3-436d-b9df-98e009e715bc.png">

In the screen shot below notice how "web logs count" and "sum of bytes" text fades. To view, install web logs sample data set. Open map that ships with web logs. Then click on circle marker in map
<img width="482" alt="Screen Shot 2022-11-16 at 1 34 34 PM" src="https://user-images.githubusercontent.com/373691/202289281-8855d58c-a80c-419e-a1b4-7fbd7f3e0235.png">

constancecchen commented:
@nreese I can repro both issues, but as a heads up, the first one (that involves `EuiFacetGroup`/`EuiFacetButton` was _not_ touched as part of this upgrade/changelog (but was in the previous upgrade). Are you confident this issue isn't already appearing in latest `main`? If it is, I would want to open the fix as a separate PR, which would need to be backported in for 8.6 FF, and not as part of the upgrade PR.

mdefazio reviewed (approved):
Thank you! Curious to see how we can make this easier (for you) next time!

nreese commented:
> Are you confident this issue isn't already appearing in latest main?

Thanks for pointing this out. Both issues appear on main and are not introduced with this PR.

nreese reviewed (approved):
kibana-gis changes LGTM
code review, tested in chrome

constancecchen commented:
@nreese Haha, I was just mention that I could repro on main. Y'all will need a separate backport for 8.6 FF. I'm looking into the fixes on EUI's side of things and then will open an 8.6 FF backport as well as a backport into this PR. Thank you for your eagle eye, definitely glad we're not shipping a major with those visual bugs!

constancecchen commented:
> Thank you! Curious to see how we can make this easier (for you) next time!

No problem at all - all your eyes & QA is much appreciated as EUI generally doesn't have the depth of knowledge you all do regarding your respective apps. In terms of helpfulness, we always appreciate 1. links to the problem area (either the URL where the issue occurs, or the file with the problematic code), and 2. repro steps, if getting to the view with the bug takes special knowledge or local data that doesn't come OOTB with the Kibana sample data.

cee-chen committed:
[Maps] Fix broken facet button display

cee-chen committed:
[EuiButtonDisplay] snapshot updates

cee-chen committed:
[EuiFlyout] snapshot updates

- caused by overflow shadow CSS changing

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

cee-chen committed:
Update EUI to v70.2.2 to get latest backport fixes

constancecchen commented:
@nreese Fixes for the two issues should now be in this branch, if you don't mind pulling down to take a look/re-QA! The 8.6 backport is here: https://github.com/elastic/kibana/pull/145490

stratoula commented:
@constancecchen thanx! It is fixed but they still seem different.
<img width="412" alt="image" src="https://user-images.githubusercontent.com/17003240/202389574-91bc8862-6f8a-4e47-92da-22071c91bd7c.png">


On main they don't have this big space between them and I prefer to keep it that way.

<img width="390" alt="image" src="https://user-images.githubusercontent.com/17003240/202389630-7581b72d-e538-4a06-927d-2dfdd692aba1.png">

jughosta reviewed (commented):
Hi @constancecchen ! Noticed some issues, mostly in mobile view on Discover page:

- Mobile layout seems to have too much empty space and scrolling is affected
<img width="300" alt="Screenshot 2022-11-17 at 09 56 08" src="https://user-images.githubusercontent.com/1415710/202403364-31e342f4-c4c1-49d6-b85a-036aef63bb82.png">

- Fields flyout is not scrollable
<img width="300" alt="Screenshot 2022-11-17 at 09 56 57" src="https://user-images.githubusercontent.com/1415710/202403556-be907109-d56c-411e-8e45-cac244ca66a3.png">

- "Create field" flyout does not fit
<img width="300" alt="Screenshot 2022-11-17 at 09 55 39" src="https://user-images.githubusercontent.com/1415710/202403440-774f7159-0053-48a1-b250-d330fa9ecf5a.png">

- Time picker had a denser content before, without scrolling
<img width="300" alt="Screenshot 2022-11-17 at 10 01 37" src="https://user-images.githubusercontent.com/1415710/202403808-c5e17715-fee5-4263-9ce3-b2559e6ae74f.png">

- Advanced Settings heading is a bit off
<img width="300" alt="Screenshot 2022-11-17 at 09 57 53" src="https://user-images.githubusercontent.com/1415710/202403951-efe6eb9e-75ef-4b23-8487-868ac7504881.png">

stratoula commented:
Moreover I see a big padding on the SuperDatePicker, check the commonly used section while the Recently used has a different padding

<img width="417" alt="image" src="https://user-images.githubusercontent.com/17003240/202411203-a597e73e-e202-4ef4-a684-9363c07e0d43.png">

peteharverson reviewed (approved):
Thanks for fixing the flex issues in ML @constancecchen! Re-tested and ML changes LGTM

cee-chen committed:
[Maps] snapshots

elizabetdev committed:
Fixing `Lens` dimension editor operation button size

miukimiu commented:
@stratoula I pushed b6f322a to fix the issue with https://github.com/elastic/kibana/pull/144845#issuecomment-1318236026

The other issue with the **SuperDatePicker** needs to be fixed in EUI.

nreese commented:
> Fixes for the two issues should now be in this branch, if you don't mind pulling down to take a look/re-QA! The 8.6 backport is here: https://github.com/elastic/kibana/pull/145490

Thank you. Just tested and verified issues are resolved.

cee-chen committed:
Upgrade EUI to v70.2.4

- To get EuiSuperDatePicker and EuiFlex fixes

cee-chen committed:
[Advanced Settings] Fix flex gap/margins

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

constancecchen commented:
Addressing @jughosta's comments:

> Mobile layout seems to have to much empty space and scrolling is affected
> Advanced Settings heading is a bit off

Thank you for catching this! 🦅 These were both the same underlying bug with nested EuiFlexGroups (https://github.com/elastic/eui/pull/6381), which a fix for should have just been backported in.

> Fields flyout is not scrollable

Flyouts were not significantly touched in this upgrade, meaning this is almost certainly already an issue on main. The source of this breakage would have come from when EuiFlyout was converted to Emotion (which merged in in the last EUI upgrade PR). It looks like this is the culprit:

https://github.com/elastic/kibana/blob/fcdcc2d021d37adedf6f746281252d6c599abb6b/src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx#L323-L324

The `.euiFlyoutBody` CSS no longer has any styles attached to it since its Emotion conversion. This will need an 8.6FF backport since it's already in main. I'll open up a separate PR, I don't think this should be fixed in this upgrade.

> "Create field" flyout does not fit

This flyout does not appear to use our standard flyout layout / its content is fairly custom. This is an issue with implementation, not with EUI or this upgrade - I can help resolve it in a separate PR as well, and will ping you when I open that.

> Time picker had a denser content before, without scrolling

Addressed via EUI backport - EuiSuperDatePicker issues should be fixed in this branch now!

Thank you all again for your super thorough QA, it's very much appreciated! 🙇

constancecchen commented:
Pinging remaining CODEOWNERS: @elastic/security-solution-platform @elastic/protections-experience  @elastic/security-onboarding-and-lifecycle-mgt @ashokaditya @dasansol92

Last call for review/approval/QA - I'll be asking KibanaOps to admin merge this by the end of the week. This PR will not be landing in 8.6, so any issues found will have the full 8.7 minor to resolve.

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-70.2.x

stratoula reviewed (approved):
Thanx @constancecchen for your amazing work here 🙇 Visualizations team LGTM

ashokaditya commented on x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/use_response_actions_log_table.tsx:
```diff
@@ -69,7 +69,7 @@ const StyledDescriptionList = euiStyled(EuiDescriptionList).attrs({
 // output section styles
 const topSpacingCss = css`
-  ${(props) => `${props.theme.eui.euiCodeBlockPaddingModifiers.paddingMedium} 0`}
+  ${(props) => `${props.theme.eui.euiSize} 0`}
```
Not related to this line of change, but for some reason, the `font-weight` in the description list (`customDescriptionListCss` lines 48-61) is indeed heavier (semiBold). Perhaps `euiFontWeightSemiBold` was being overwritten by something else in previous versions and with this update, it isn't.


Basically, the weight should be normal and the list should be seen like so,

![Screenshot 2022-11-18 at 10 24 34](https://user-images.githubusercontent.com/1849116/202671337-ce049d0d-6342-4bd1-984b-88ba238ba7b6.png)

Yet, in this update branch, it shows up like this:

![Screenshot 2022-11-18 at 09 50 12](https://user-images.githubusercontent.com/1849116/202671441-59b92f33-5e82-44c0-b31a-bc18c44ee662.png)


If you replace lines 48-61 above, with the following it should fix that.

```
const customDescriptionListCss = css`
  &.euiDescriptionList {
    > .euiDescriptionList__title {
      color: ${(props) => props.theme.eui.euiColorDarkShade};
      font-size: ${(props) => props.theme.eui.euiFontSizeXS};
    }

    > .euiDescriptionList__title,
    .euiDescriptionList__description {
      font-weight: ${(props) => props.theme.eui.euiFontWeightRegular};
      margin-top: ${(props) => props.theme.eui.euiSizeS};
    }
  }
`;
```

constancecchen replied:
Addressed in https://github.com/elastic/kibana/pull/144845/commits/d5f93d327e153105f57dabe1d981ee8ce3404527 - thanks for the find and the quick fix!

Yes, in our Emotion conversions, the majority of our CSS should now be much flatter and have less nesting/specificity, so CSS that was previously _not_ being overridden may now start overriding our CSS. However, the upside is it should be slightly less painful to override EUI CSS in the future.

constancecchen replied:
@ashokaditya It's worth noting that `EuiDescriptionList` was not converted to Emotion in this EUI upgrade, meaning that this bug has been around for a while. Looking back at our changelogs, it looks like the Emotion conversion of `EuiDescriptionList` made it into 8.5, so you will need to backport to both 8.5 and 8.6 if the visual bug bothers you enough to want to fix it in past releases.

ashokaditya reviewed (commented):
Thanks for this huge update. 🙇 . `security_solution/management` changes LGTM. I checked out the changes locally and I do see some discrepancies with font weight for the response actions history. I'll test it again after the suggested change goes in. Thanks 🙏

jughosta commented:
Filter icon is different now and looks smaller in comparison with others. Is it expected?

Discover sidebar > field popover:
(before)
<img width="150" alt="Screenshot 2022-11-18 at 13 51 05" src="https://user-images.githubusercontent.com/1415710/202709799-1d70a5d7-c2dc-444b-bb74-5cff9aefae03.png">
(after)
<img width="154" alt="Screenshot 2022-11-18 at 13 49 50" src="https://user-images.githubusercontent.com/1415710/202709861-f1fe8593-1d3a-4d7a-bc25-64e8e00b824a.png">

Discover grid > Doc viewer:
(before)
<img width="254" alt="Screenshot 2022-11-18 at 13 51 19" src="https://user-images.githubusercontent.com/1415710/202710408-b22166dd-9785-410c-96eb-5bdbd9c0c478.png">
(after)
<img width="335" alt="Screenshot 2022-11-18 at 13 50 08" src="https://user-images.githubusercontent.com/1415710/202710501-834aebbf-ba40-4a1d-b844-96de70c593b4.png">

cee-chen committed:
[Security] EuiDescriptionList tweaks

- per request

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-70.2.x

constancecchen commented:
> Filter icon is different now and looks smaller in comparison with others. Is it expected?

@jughosta Yes, this was an intentional EUI designer change as confirmed by @miukimiu above. https://github.com/elastic/eui/pull/6282

cc @MichaelMarcialis - I do agree that the filter icon now does look a little odd next to the +/- icons, so you may want to consider re-adding a `filterInCircle` icon?

chandlerprall reviewed (approved):
Changes LGTM! ❤️

miukimiu commented:
@jughosta I'll re-include the old `filter` icon as `filterInCircle ` in EUI. 

CC @MichaelMarcialis

constancecchen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-70.2.x
